### PR TITLE
feat: LT-1670 - Session timeout

### PR DESCRIPTION
**Pull request by a-bek**

### CODE REVIEW

File: conf/settings.py [hunk 1]
```diff
@@ -222,3 +223,7 @@
 CSP_SCRIPT_SRC = env("CSP_SCRIPT_SRC")
 CSP_FONT_SRC = env("CSP_FONT_SRC")
 CSP_REPORT_ONLY = env("CSP_REPORT_ONLY")
+
+
+# Session timeout
+SESSION_EXPIRE_SECONDS = 3600
```
Comment (Mark-Spiers-kainos) [inline conf/settings.py]: Same as exporter should this not be 30 minutes? and I think I prefer it shown as `x * x`

**Comment by BenAndrew-Kainos** [comment]
Please address all the same actions as on exporter (then I'll approve)

File: conf/middleware.py [hunk 2]
```diff
@@ -59,3 +62,26 @@ def __call__(self, request):
...
+
+
+class SessionTimeoutMiddleware:
+    def __init__(self, get_response=None):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        start = request.session.get(SESSION_TIMEOUT_KEY, time.time())
+
+        timeout = getattr(settings, "SESSION_EXPIRE_SECONDS", 60 * 60)  # Defaults to 60 minutes
```
Comment (BenAndrew-Kainos) [inline conf/middleware.py]: Move this value (60 * 60) to a constant

File: conf/middleware.py [hunk 3]
```diff
@@ -59,3 +62,26 @@ def __call__(self, request):
...
+
+        # Expire the session if more than start time + timeout time has occurred
+        if time.time() - start > timeout:
+            request.session.flush()
+            logout(request)
+            return redirect(env("AUTHBROKER_URL") + "/logout/")
```
Comment (BenAndrew-Kainos) [inline conf/middleware.py]: Please define URL as a constant

**Pull request merged**